crates/typst-library/src/meta/figure.rs

```diff
@@ -498,6 +502,32 @@ pub struct FigureCaption {
     pub location: Option<Location>,
 }
 
+impl FigureCaption {
+    pub fn get_separator(&self, styles: StyleChain) -> Content {
+        match self.separator(styles) {
+            Smart::Auto => {
+                let region = TextElem::region_in(styles);
+                let separator = match TextElem::lang_in(styles) {
+                    Lang::CHINESE => "：",
+                    Lang::FRENCH if option_eq(region, "CH") => "\u{202f}: ",
+                    Lang::FRENCH => "\u{a0}: ",
+                    Lang::DANISH
+                    | Lang::DUTCH
+                    | Lang::ENGLISH
+                    | Lang::GERMAN
+                    | Lang::ITALIAN
+                    | Lang::RUSSIAN
+                    | Lang::SPANISH
+                    | Lang::SWEDISH
+                    | _ => ": ",
+                };
+                TextElem::packed(separator)
+            }
```

Comment:
I think the auto branch should be a free-standing function that takes lang and region and returns a `&'static str` like `LocalName` does. And then maybe the separator could be synthesized into its resolved versino (like `kind` is), so that user show rules get the correct one.